### PR TITLE
Remove java-legacy client from the codebase and build process

### LINKED ISSUE #8389: Remove java legacy client
Sub-task of #8193 

Remove the java-legacy client code and any mention of it from our build process:
- Makefile
- Github actions

### PR DESCRIPTION
**Pull request by yonipeleg33**
Closes #8389

# :relieved:

### CODE REVIEW

**Review by idanovo — APPROVED** [review]
Yesss

**Comment by yonipeleg33** [comment]
I'd like to get an OK from @iddoavn and @treeverse/product before merging this change

**Pull request merged**
